### ISSUE TITLE
Bookkeeping Loss Variables

### ISSUE THREAD

**Issue by alansun17904** [alansun17904/MAAA#22]
## Summary

Go through the MeZO training code and the computation of the loss function. When the loss value begins to go negative, trace the variables that compose the loss function and check their values. In this way, we would be able to nail down where the negative loss is coming from. 

## Acceptance Criteria

For a given training run and step, when the loss goes negative list out the variables used to compute the loss function and their associated values. The values of these variables in combinations with the equation of the loss should reconstruct this very negative loss value.
